gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/nfp_node_driver.py

```diff
@@ -727,35 +824,58 @@ def _create_network_function(self, context):
                 context.core_plugin.update_port(
                     context.plugin_context, provider_port['id'], port)
 
-        port_info = []
-        if service_targets.get('provider_pts'):
-            # Device case, for Base mode ports won't be available.
-            port_info = [
-                {
-                    'id': service_targets['provider_pts'][0],
-                    'port_model': nfp_constants.GBP_PORT,
-                    'port_classification': nfp_constants.PROVIDER,
-                }
-            ]
-        if service_targets.get('consumer_ports'):
-            port_info.append({
-                'id': service_targets['consumer_pts'][0],
-                'port_model': nfp_constants.GBP_PORT,
-                'port_classification': nfp_constants.CONSUMER,
-            })
-        network_function = {
-            'tenant_id': context.provider['tenant_id'],
-            'service_chain_id': sc_instance['id'],
-            'service_id': context.current_node['id'],
-            'service_profile_id': context.current_profile['id'],
-            'management_ptg_id': sc_instance['management_ptg_id'],
+        provider = {
+            'pt': service_targets.get('provider_pt_objs', [None])[0],
+            'ptg': service_targets.get('provider_ptg', [None])[0],
+            'port': service_targets.get('provider_ports', [None])[0],
+            'subnet': service_targets.get('provider_subnet', None),
+            'port_model': nfp_constants.GBP_PORT,
+            'port_classification': nfp_constants.PROVIDER}
+
+        consumer_pt = None
+        consumer_ptg = None
+        consumer_ports = None
+
+        if service_targets['consumer_pt_objs']:
+            consumer_pt = service_targets.get('consumer_pt_objs', [None])[0]
```

Comment:
None never get executed, since we already checked with if.
Why we are doing different for both provider, consumer, Can we do in similar way